src/react-integration/__tests__/integration.tsx

```diff
@@ -1,202 +1,164 @@
-import React, { Suspense } from 'react';
-import { render } from 'react-testing-library';
-import { cleanup, renderHook, act } from 'react-hooks-testing-library';
+import React from 'react';
+import { cleanup } from 'react-hooks-testing-library';
 import nock from 'nock';
 
 import { CoolerArticleResource, UserResource } from '../../__tests__/common';
 import { useResource, useFetcher } from '../hooks';
-import RestProvider from '../provider';
-import NetworkManager from '../../state/NetworkManager';
-import { FetchAction, ReceiveAction } from '../../types';
-
-class MockNetworkManager extends NetworkManager {
-  handleFetch(action: FetchAction, dispatch: React.Dispatch<any>) {
-    const mockDispatch: typeof dispatch = (v: any) => {
-      act(() => {
-        dispatch(v);
-      });
-    };
-    return super.handleFetch(action, mockDispatch);
-  }
-  handleReceive(action: ReceiveAction) {
-    act(() => {
-      super.handleReceive(action);
-    });
-  }
-}
+import createRenderRestHook from '../../test/helper';
+import {
+  makeRestProvider,
+  makeExternalCacheProvider,
+} from '../../test/providers';
 
 afterEach(() => {
   cleanup();
 });
 
-describe('<RestProvider />', () => {
-  const payload = {
-    id: 5,
-    title: 'hi ho',
-    content: 'whatever',
-    tags: ['a', 'best', 'react'],
-  };
-  const users = [
-    {
-      id: 23,
-      username: 'bob',
-      email: 'bob@bob.com',
-      isAdmin: false,
-    },
-    {
-      id: 7342,
-      username: 'lindsey',
-      email: 'lindsey@bob.com',
-      isAdmin: true,
-    },
-  ];
-  const nested = [
-    {
+for (const makeProvider of [makeRestProvider, makeExternalCacheProvider]) {
```

Comment:
this tabbed over most of the file making it look at way more changed. truthfully only the use of createRenderRestHook was changed